test/Microsoft.AspNet.Mvc.TagHelpers.Test/ScriptTagHelperTest.cs

```diff
@@ -0,0 +1,255 @@
+        [Theory]
+        [InlineData("~/blank.js")]
+        [InlineData(null)]
+        public async Task RunsWhenRequiredAttributesArePresent(string srcValue)
+        {
+            // Arrange
+            var attributes = new Dictionary<string, object>
+                {
+                    { "asp-fallback-src", "http://www.example.com/blank.js" },
+                    { "asp-fallback-test", "isavailable()" },
+                };
+
+            if (srcValue != null)
+            {
+                attributes.Add("src", srcValue);
+            }
+
+            var context = MakeTagHelperContext(attributes);
+
+            var output = MakeTagHelperOutput("script");
+            var logger = new Logger();
+
+            // Act
+            var helper = new ScriptTagHelper()
+            {
+                Logger = logger,
+                FallbackSrc = "http://www.example.com/blank.js",
+                FallbackTestExpression = "isavailable()",
+            };
+
+            await helper.ProcessAsync(context, output);
+
+            // Assert
+            Assert.Null(output.TagName);
+            Assert.NotNull(output.Content);
+            Assert.True(output.ContentSet);
+            Assert.Equal(0, logger.Logged.Count);
+        }
+
+        [Theory]
+        [InlineData("asp-fallback-src")]
+        [InlineData("asp-fallback-test")]
+        public async Task DoesNotRunWhenARequiredAttributeIsMissing(string attributeToRemove)
+        {
+            // Arrange
+            var attributes = new Dictionary<string, object>
+                {
+                    { "asp-fallback-src", "http://www.example.com/blank.js" },
+                    { "asp-fallback-test", "isavailable()" },
+                };
+
+            attributes.Remove(attributeToRemove);
+
+            Assert.Equal(1, attributes.Count);
+
+            var context = MakeTagHelperContext(attributes);
+
+            var output = MakeTagHelperOutput("script");
+            var logger = new Logger();
+
+            // Act
+            var helper = new ScriptTagHelper
+            {
+                Logger = logger,
+            };
+
+            await helper.ProcessAsync(context, output);
+
+            // Assert
+            Assert.NotNull(output.TagName);
+            Assert.False(output.ContentSet);
+        }
+
+        [Fact]
+        public async Task DoesNotRunWhenAllRequiredAttributesAreMissing()
+        {
+            // Arrange
+            var context = MakeTagHelperContext();
+            var output = MakeTagHelperOutput("script");
+            var logger = new Logger();
+
+            // Act
+            var helper = new ScriptTagHelper
+            {
+                Logger = logger,
+            };
+
+            await helper.ProcessAsync(context, output);
+
+            // Assert
+            Assert.NotNull(output.TagName);
+            Assert.False(output.ContentSet);
+        }
+
+        [Fact]
+        public async Task LogsWhenARequiredAttributeIsMissing()
```

Comment:
/cc @DamianEdwards We should probably add some log tests since it's essentially a first class concept that we're trying to incorporate into all of our `TagHelper`s